Tests/ApodiniTests/PropertiesTests/BindingTests.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  BindingTests.swift
+//  
+//
+//  Created by Max Obermeier on 24.02.21.
+//
+
+import Foundation
+
+@testable import Apodini
+import XCTest
+import XCTApodini
+import ApodiniREST
+
+final class BindingTests: ApodiniTests, EnvironmentAccessible {
+    struct Greeter: Handler {
+        @Binding var country: String?
+        
+        func handle() -> String {
+            country ?? "World"
+        }
+    }
+    
+    struct ReallyOptionalGreeter: Handler {
+        @Binding var country: String??
+        
+        func handle() -> String? {
+            if let country = self.country {
+                return country ?? "World"
+            } else {
+                return nil
+            }
+        }
+    }
+        
+    var featured: String? = "Italy"
+    
+    @PathParameter var selectedCountry: String
+    @Environment(\BindingTests.featured) var featuredCountry
+    
+    @ComponentBuilder
+    var testService: some Component {
+        Greeter(country: .constant(nil))
+        Group("default") {
+            Greeter(country: Binding<String?>(.constant("USA")))
+        }
+        Group("optional") {
+            Greeter(country: Parameter<String?>().projectedValue)
+        }
+        Group("featured") {
+            Greeter(country: $featuredCountry)
+        }
+        Group("country", $selectedCountry) {
+            Greeter(country: Binding<String?>($selectedCountry.projectedValue))
```

Comment:
It would be cool to think about a way to make this a bit cleaner. Maybe a `binding` or an `optionalBinding` property that does the same but hides it behind a property of a `@Parameter` or `@PathParameter` so we could write:
```suggestion
            Greeter(country: $selectedCountry.projectedValue.optionalBinding)
```
or even better:
```suggestion
            Greeter(country: $selectedCountry.optionalBinding)
```
I could imagine that could make the `public init<V: Codable>(_ base: Binding<V>) where Value == V? { /* ... */}` initializer internal in this process.

Reply:
I added an `binding` computed property on `@PathParameter` and an `asOptional` one on `Binding`. This makes it a lot cleaner indeed: `_selectedCountry.binding.asOptional`.

> I could imagine that could make the public init<V: Codable>(_ base: Binding<V>) where Value == V? { /* ... */} initializer internal in this process.

I See no point in doing so. The signature is an exact copy of SwiftUI's `Binding`. I think having a familiar element there might help some developers.

Reply:
Looks good 🚀